relay-quotas/src/redis.rs

```diff
@@ -797,4 +835,58 @@ mod tests {
             vec![false]
         );
     }
+
+    fn redis_quota_for_metric_buckets(window: u64, limit: u64) -> RedisQuota<'static> {
+        let quota = Quota {
+            id: Some("foo".to_owned()),
+            categories: DataCategories::new(),
+            scope: QuotaScope::Global,
+            scope_id: None,
+            window: Some(window),
+            limit: Some(limit),
+            reason_code: None,
+        };
+
+        let inner_scoping = Scoping {
+            organization_id: 69420,
+            project_id: ProjectId::new(42),
+            project_key: ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap(),
+            key_id: Some(4711),
+        };
+
+        let static_inner_scoping: &'static Scoping = Box::leak(Box::new(inner_scoping));
+        let static_quota: &'static Quota = Box::leak(Box::new(quota));
+
+        let scoping = ItemScoping {
+            category: DataCategory::MetricBucket,
+            scoping: static_inner_scoping,
+        };
+
+        RedisQuota::new(static_quota, scoping, UnixTimestamp::now()).unwrap()
+    }
+
+    #[test]
+    fn test_global_ratelimit() {
```

Comment:
We should also have a test using at least two separate `GlobalRateLimits` accessing the same Redis and making sure they are correct. This doesn't necessarily allow us to test perfect parallelism but should allow us to test multiple (hundreds) of rate limit attempts through multiple limiters and verify at the end that the total count is correct.